zeppelin-server/src/main/java/org/apache/zeppelin/realm/ActiveDirectoryGroupRealm.java

```diff
@@ -291,11 +291,11 @@ protected AuthorizationInfo buildAuthorizationInfo(Set<String> roleNames) {
     SearchControls searchCtls = new SearchControls();
     searchCtls.setSearchScope(SearchControls.SUBTREE_SCOPE);
     String userPrincipalName = username;
-    if (this.principalSuffix != null && userPrincipalName.indexOf('@') < 0) {
-      userPrincipalName += principalSuffix;
+    if (this.principalSuffix != null && userPrincipalName.indexOf('@') > 1) {
+      userPrincipalName = userPrincipalName.split("@")[0];
     }
 
-    String searchFilter = "(&(objectClass=*)(userPrincipalName=" + userPrincipalName + "))";
+    String searchFilter = "(&(objectClass=*)(sAMAccountName=" + userPrincipalName + "))";
```

Comment:
This `(userPrincipalName=?)`  is used in two places `public List<String> searchForUserName` and `private Set<String> getRoleNamesForUser`.
Also, can you make to property configurable like https://github.com/apache/zeppelin/blob/master/zeppelin-server/src/main/java/org/apache/zeppelin/realm/LdapRealm.java#L107

Reply:
sorry @prabhjyotsingh for the late reply, you were right, the other search was used for searching when setting note permissions and i fixed that one also
also userSearchParameter is configurable via shiro.ini from now but uses default if not set

please let me know i i should update doc somewhare as I was not able to find any resources